pkg/operator/controllers/configs/config_autonodesize.go

```diff
@@ -0,0 +1,107 @@
+// Auto Node sizing tells machine-config-operator to calculate
+// system reserved dynamically based on the node physical size.
+// The config is applied to all worker nodes.
+// More info in the doc:
+// - https://docs.openshift.com/container-platform/4.8/nodes/nodes/nodes-nodes-resources-configuring.html
+package configs
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/coreos/ignition/v2/config/util"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+const (
+	configName = "dynamic-node"
+)
+
+type autoNodeSizeConfig struct {
+}
+
+func NewAutoNodeSizeConfig() Config {
+	return &autoNodeSizeConfig{}
+}
+
+func (c *autoNodeSizeConfig) Name() string {
+	return "AutoNodeSize config"
+}
+
+func (c *autoNodeSizeConfig) IsApplicable(aro arov1alpha1.Cluster, r *Reconciler, ctx context.Context) bool {
+	var config mcv1.KubeletConfig
+	var err error
+
+	key := types.NamespacedName{
+		Name: configName,
+	}
+
+	if aro.Spec.Features.ReconcileAutoSizedNodes {
```

Comment:
It is `false` by default. Do we want to update `pkg/operator/deploy/deploy.go` to enable it?